From 540bfae9e702f95096220c9a604c0d463048e96f Mon Sep 17 00:00:00 2001
From: Mohammad Athari Bin Ismail <mohammad.athari.ismail@intel.com>
Date: Thu, 5 Nov 2020 17:13:40 +0800
Subject: [PATCH] net: stmmac: Start phylink after DW xPCS initialization

Move phylink_start() after setup_phy_conv() in stmmac_open(). This a
proper sequence where DW xCPS should be initialized prior to
phylink start.

Signed-off-by: Mohammad Athari Bin Ismail <mohammad.athari.ismail@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 5086a9b80feb..4a8165897eaf 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -3610,8 +3610,6 @@ static int stmmac_open(struct net_device *dev)
 
 	stmmac_init_coalesce(priv);
 
-	phylink_start(priv->phylink);
-
 	ret = stmmac_request_irq(dev);
 	if (ret)
 		goto irq_error;
@@ -3637,6 +3635,8 @@ static int stmmac_open(struct net_device *dev)
 		}
 	}
 
+	phylink_start(priv->phylink);
+
 	stmmac_enable_all_queues(priv);
 	netif_tx_start_all_queues(priv->dev);
 
-- 
2.17.1

